goal_src/jak2/engine/target/gun/gun-util.gc

```diff
@@ -5,5 +5,1180 @@
+      )
+    (let ((v1-30 (-> (the-as target gp-0) draw color-emissive quad)))
+      (set! (-> (the-as gun self) draw color-emissive quad) v1-30)
+      )
+    (set! (-> (the-as gun self) draw force-fade) (-> (the-as target gp-0) draw force-fade))
+    (set! (-> (the-as gun self) draw death-vertex-skip)
+          (-> (the-as target (-> (the-as gun self) parent 0)) draw death-vertex-skip)
+          )
+    (set! (-> (the-as gun self) draw death-effect)
+          (-> (the-as target (-> (the-as gun self) parent 0)) draw death-effect)
+          )
+    (set! (-> (the-as gun self) draw death-timer)
+          (-> (the-as target (-> (the-as gun self) parent 0)) draw death-timer)
+          )
+    (set! (-> (the-as gun self) draw death-timer-org)
+          (-> (the-as target (-> (the-as gun self) parent 0)) draw death-timer-org)
+          )
+    (set! (-> (the-as gun self) draw death-draw-overlap)
+          (-> (the-as target (-> (the-as gun self) parent 0)) draw death-draw-overlap)
+          )
+    (let ((v1-55 (-> (the-as target gp-0) draw shadow-ctrl settings shadow-dir quad)))
+      (set! (-> (the-as gun self) draw shadow-ctrl settings shadow-dir quad) v1-55)
+      )
+    (let ((t9-4 (method-of-object (-> (the-as gun self) barrel) joint-mod-method-13)))
+      (-> (the-as target gp-0) gun fire-spin)
+      (t9-4)
+      )
+    (if (or (logtest? (-> (the-as target gp-0) draw status)
+                      (draw-control-status no-draw no-draw-temp no-draw-bounds no-draw-bounds2)
+                      )
+            (zero? (-> (the-as target gp-0) skel active-channels))
+            )
+        (logior! (-> (the-as gun self) draw status) (draw-control-status no-draw))
+        (logclear! (-> (the-as gun self) draw status) (draw-control-status no-draw))
+        )
+    (if (logtest? (-> (the-as target gp-0) draw status) (draw-control-status force-fade))
+        (logior! (-> (the-as gun self) draw status) (draw-control-status force-fade))
+        (logclear! (-> (the-as gun self) draw status) (draw-control-status force-fade))
+        )
+    )
+  (dotimes (gp-1 4)
+    (cond
+      ((= (-> (the-as target (-> (the-as gun self) parent 0)) game gun-ammo gp-1) 0.0)
+       (when (!= (-> (the-as gun self) mag-scale gp-1) 0.0)
+         (let ((s5-2 (new 'stack-no-clear 'matrix)))
+           (let ((s3-0 (new 'stack-no-clear 'vector))
+                 (s4-1 (new 'stack-no-clear 'vector))
+                 )
+             (vector<-cspace! s3-0 (-> (the-as gun self) mag gp-1 joint))
+             (vector-normalize-copy!
+               s4-1
+               (the-as vector (&-> (-> (the-as gun self) mag gp-1 joint bone) transform data 4))
+               40960.0
+               )
+             (set! (-> s4-1 y) (+ 81920.0 (-> s4-1 y)))
+             (set! (-> s5-2 data 0) (the-as float (-> (the-as gun self) entity)))
+             (set! (-> s5-2 data 1) 1.0)
+             (s.d! (+ s5-2 16) 0)
```

Comment:
hm, I don't think this will actually compile successfully. Maybe for now we should leave out this file - I think there's a bunch of stuff in here that's not handled well yet.  But just having the types filled out in `all-types.gc` will still be helpful for when we finally get to this file

Reply:
Emptied the contents of `gun-util.gc` for now